cmd/main.go

```diff
@@ -39,15 +39,20 @@ import (
 func main() {
 	defaultLog := logging.GetDefaultLogger()
 
-	defaultConfig := config.GetDefaultAppConfig(defaultLog)
-	if err := cli.LoadFlags(defaultConfig); err != nil {
+	if err := config.SetDefaults(defaultLog); err != nil {
```

Comment:
Isn't this method called when we do "Load" in this line?
`cfg, log, err := config.Load(defaultCfg)`

Reply:
I removed it from `Load` method because it has to be called before `LoadFlags` and `LoadFlags` cannot be called inside config because of import cycle

Reply:
Ok, understand

Reply:
Sorry to unresolve this, but I just have one more idea. What if we move `flags.go` back from `/cli` to `/config`? 

It's a nice idea to put them in the separate package, but I don't see a good enough reason to have cli flags away from config (after all, flags are config-related as well) and it creates unnecessary complications, such as this.

I would much rather have flags.go in the config package and remove `SetDefaults()` from `main.go` and call it in the config. WDYT?

Reply:
okay, I missed that, nevermind then